<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SeqSrchViewClass</class>
 <widget class="QWidget" name="SeqSrchViewClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1097</width>
    <height>530</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QPlainTextEdit" name="seqSrchPlainTextEdit">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>130</y>
     <width>371</width>
     <height>171</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="seqSrchHomeBtn">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>10</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Home</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="seqSrchInfoTextBrowser">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>330</y>
     <width>371</width>
     <height>191</height>
    </rect>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Nunito','sans-serif'; font-size:8pt; color:#000000; background-color:#ffffff;&quot;&gt;Linear search is a very simple search algorithm. In this type of search, a sequential search is made over all items one by one. Every item is checked and if a match is found then that particular item is returned, otherwise the search continues till the end of the data collection.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Nunito','sans-serif'; font-size:8pt; color:#000000;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:8pt; color:#000000;&quot;&gt;The time complexity of the sequential search is linear O(n&lt;/span&gt;&lt;span style=&quot; font-size:8pt; color:#000000;&quot;&gt;) &lt;/span&gt;&lt;/p&gt;
&lt;h2 style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; line-height:125%; font-family:'Heebo','sans-serif'; font-size:8pt; font-weight:496; color:#000000;&quot;&gt;&lt;br /&gt;&lt;/h2&gt;
&lt;h2 style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; line-height:125%;&quot;&gt;&lt;span style=&quot; font-family:'Heebo','sans-serif'; font-size:8pt; font-weight:496; color:#000000;&quot;&gt;Pseudocode&lt;/span&gt;&lt;/h2&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;procedure linear_search &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;(&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;list&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;,&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;value&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Courier New'; font-size:8pt; color:#666600; background-color:#eeeeee;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;   &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;for&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; each item &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;in&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; the list&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;if&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; match item &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#666600;&quot;&gt;==&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;value&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt;         &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000088;&quot;&gt;return&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#000000;&quot;&gt; the item&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#008800;&quot;&gt;'s location&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#008800;&quot;&gt;      end if&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#008800;&quot;&gt;   end for&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Courier New'; font-size:8pt; color:#008800; background-color:#eeeeee;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#eeeeee;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt; color:#008800;&quot;&gt;end procedure&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="seqSrchConsoleLabel">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>110</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Console:</string>
   </property>
  </widget>
  <widget class="Line" name="seqSrchAnimTextAreaSeparator">
   <property name="geometry">
    <rect>
     <x>695</x>
     <y>110</y>
     <width>31</width>
     <height>411</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QWidget" name="seqSrchAnimAreaPane" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>691</width>
     <height>411</height>
    </rect>
   </property>
   <widget class="QGraphicsView" name="seqSrchAnimArea">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>701</width>
      <height>411</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="seqSrchInfoConsoleLabel">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>310</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Theory:</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>691</width>
     <height>97</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="seqSrchHeaderAndControlPane">
    <item>
     <widget class="QLabel" name="seqSrchHeaderLabel">
      <property name="layoutDirection">
       <enum>Qt::RightToLeft</enum>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt; font-weight:600;&quot;&gt;SEQUENTIAL SEARCH&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="seqSrchControlPane">
      <item>
       <widget class="QLineEdit" name="seqSrchInsertLineEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="seqSrchSrchBtn">
        <property name="text">
         <string>Search</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="seqSrchSrchNewValSepLine">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="seqSrchNewValBtn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>New Values</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
